base/strings/basic.jl

```diff
@@ -29,7 +29,7 @@ convert(::Type{Vector{UInt8}}, s::AbstractString) = bytestring(s).data
 convert(::Type{Array{UInt8}}, s::AbstractString) = bytestring(s).data
 convert(::Type{ByteString}, s::AbstractString) = bytestring(s)
 convert(::Type{Vector{Char}}, s::AbstractString) = collect(s)
-convert(::Type{Symbol}, s::AbstractString) = symbol(s)
+convert(::Type{Symbol}, s::AbstractString) = Symbol(s)
```

Comment:
@nalimilan Can you explain? [The docs on constructors](http://docs.julialang.org/en/release-0.4/manual/constructors/#constructors-call-and-conversion) seems to suggest that `Symbol(a)` and `convert(Symbol, a)` should be equivalent. By default the constructor is defined in terms of `convert`, but in this case the implementation (that I could not find) seems to be in the constructor.